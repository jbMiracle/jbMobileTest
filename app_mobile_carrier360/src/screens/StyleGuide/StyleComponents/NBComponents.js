import React, { PureComponent } from 'react';
import { View } from 'react-native';
import { connectStyle, Text, Button, H1, H2, H3 } from 'native-base';

const styles = {
  buttonContainer: {
    flex: 1,
    flexDirection: 'row',
  },
  buttonStyle: {
    alignSelf: 'stretch',
    flex: 1,
    marginBottom: 10,
    marginTop: 10,
  },
  buttonTertiaryStyle: {
    flex: 1,
    marginBottom: 10,
    marginTop: 10,
  },
  buttonTertiaryCenterStyle: {
    flex: 1,
    marginLeft: 10,
    marginRight: 10,
  },
  subHeadingText: {
    paddingTop: 20,
    textAlign: 'center',
  },
};

class NBComponents extends PureComponent {
  render() {
    return (
      <View>
        <View>
          <H1 style={ { ...styles.subHeadingText, paddingTop: 0 } }>Text Items</H1>
          <H1 style={ { paddingTop: 10 } }>H1 Text</H1>
          <H2 style={ { paddingTop: 10 } }>H2 Text</H2>
          <H3 style={ { paddingTop: 10 } }>H3 Text</H3>
          <Text style={ { paddingTop: 12 } }>Basic Text</Text>
        </View>
        <View>
          <H1 style={ { ...styles.subHeadingText, paddingTop: 0 } }>Buttons</H1>
          <H2>Primary Buttons</H2>
          <Button
            style={ styles.buttonStyle }
            block
            large
            primary
          >
            <Text>Primary</Text>
          </Button>
          <Button
            style={ styles.buttonStyle }
            block
            large
            primaryDark
          >
            <Text>Primary Dark</Text>
          </Button>
          <Button
            style={ styles.buttonStyle }
            block
            large
            primary
            disabled
          >
            <Text>Primary Disabled</Text>
          </Button>

          <H2>Secondary Buttons</H2>
          <Button
            style={ styles.buttonStyle }
            block
            large
            secondary
          >
            <Text>Secondary</Text>
          </Button>
          <Button
            style={ styles.buttonStyle }
            block
            large
            secondaryDark
          >
            <Text>Secondary Dark</Text>
          </Button>

          <H2>Side by Side Buttons</H2>
          <View style={ styles.buttonContainer }>
            <Button
              style={ styles.buttonTertiaryStyle }
              large
              block
              primary
            >
              <Text>One</Text>
            </Button>
            <Button
              style={ [styles.buttonTertiaryStyle, styles.buttonTertiaryCenterStyle] }
              large
              block
              primary
            >
              <Text>Two</Text>
            </Button>
            <Button
              style={ styles.buttonTertiaryStyle }
              large
              block
              primary
            >
              <Text>Three Wrapped</Text>
            </Button>
          </View>
        </View>
      </View>
    );
  }
}

export default connectStyle('C360.NBComponents', styles)(NBComponents);
