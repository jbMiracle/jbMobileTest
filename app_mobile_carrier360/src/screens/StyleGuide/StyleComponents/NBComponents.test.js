import React from 'react';
import renderer from 'react-test-renderer';
import NBComponents from './NBComponents';
// const AnimatedProps = require('./nodes/AnimatedProps');
// const AnimatedProps = require('../../../../node_modules/react-native/Libraries/Animated/src/nodes/AnimatedProps');

describe('NativeBase Components', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NBComponents />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
