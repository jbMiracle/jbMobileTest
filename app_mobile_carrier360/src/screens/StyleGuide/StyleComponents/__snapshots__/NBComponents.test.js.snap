// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NativeBase Components renders correctly 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 27,
            "lineHeight": 32,
          },
          Object {
            "paddingTop": 0,
            "textAlign": "center",
          },
        ]
      }
      virtual={undefined}
    >
      Text Items
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 27,
            "lineHeight": 32,
          },
          Object {
            "paddingTop": 10,
          },
        ]
      }
      virtual={undefined}
    >
      H1 Text
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 24,
            "lineHeight": 27,
          },
          Object {
            "paddingTop": 10,
          },
        ]
      }
      virtual={undefined}
    >
      H2 Text
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 21,
            "lineHeight": 22,
          },
          Object {
            "paddingTop": 10,
          },
        ]
      }
      virtual={undefined}
    >
      H3 Text
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "paddingTop": 12,
          },
        ]
      }
      uppercase={false}
      virtual={undefined}
    >
      Basic Text
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 27,
            "lineHeight": 32,
          },
          Object {
            "paddingTop": 0,
            "textAlign": "center",
          },
        ]
      }
      virtual={undefined}
    >
      Buttons
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "lineHeight": 27,
        }
      }
      virtual={undefined}
    >
      Primary Buttons
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 2,
          "flex": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 22,
            "lineHeight": 32,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
        virtual={undefined}
      >
        Primary
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 2,
          "flex": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 22,
            "lineHeight": 32,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
        virtual={undefined}
      >
        Primary Dark
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#b5b5b5",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 2,
          "flex": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 22,
            "lineHeight": 32,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
        virtual={undefined}
      >
        Primary Disabled
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "lineHeight": 27,
        }
      }
      virtual={undefined}
    >
      Secondary Buttons
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 2,
          "flex": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 22,
            "lineHeight": 32,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
        virtual={undefined}
      >
        Secondary
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 2,
          "flex": 1,
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 22,
            "lineHeight": 32,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
        virtual={undefined}
      >
        Secondary Dark
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "lineHeight": 27,
        }
      }
      virtual={undefined}
    >
      Side by Side Buttons
    </Text>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#007aff",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flex": 1,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginTop": 10,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 22,
              "lineHeight": 32,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          One
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#007aff",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flex": 1,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 10,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 22,
              "lineHeight": 32,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Two
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#007aff",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "elevation": 2,
            "flex": 1,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginTop": 10,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 22,
              "lineHeight": 32,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
          virtual={undefined}
        >
          Three Wrapped
        </Text>
      </View>
    </View>
  </View>
</View>
`;
